x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/rest/RestSqlTestCase.java

```diff
@@ -1159,7 +1159,8 @@ public void testBinaryFieldFiltering() throws IOException {
 
     public void testPreventedUnsignedLongMaskedAccess() throws IOException {
         loadUnsignedLongTestData();
-        Version version = VersionUtils.randomVersionBetween(random(), null, VersionUtils.getPreviousVersion(INTRODUCING_UNSIGNED_LONG));
+        var preVersionCompat = SqlVersions.getAllVersions().stream().filter(v -> v.onOrAfter(INTRODUCING_UNSIGNED_LONG) == false).toList();
```

Comment:
Could be `v.before(INTRODUCING_UNSIGNED_LONG)` instead

Reply:
Ah, `SqlVersion` doesn't implement `VersionId`. If it implements `VersionId<T>` then this method will come for free

Reply:
Right, there's no `.before()` method and `SqlVersion` can't implement `VersionId<T>`, since it's in a package very light in dependencies.